ykcompile/src/lib.rs

```diff
@@ -225,65 +261,158 @@ impl<TT> TraceCompiler<TT> {
         Ok(())
     }
 
+    /// Get the type of a place.
+    fn place_ty(&self, p: &Place) -> &Ty {
+        let base_ty = SIR.ty(&self.local_decls[&p.local].ty);
+        match p {
+            Place {
+                local: _,
+                projection,
+            } if projection.is_empty() => base_ty,
+            Place {
+                local: _,
+                projection,
+            } => {
+                // FIXME this is just hacked in for now. There should be a loop which resolves
+                // arbitrarily long chains of projections.
+                assert_eq!(projection.len(), 1);
+                match projection[0] {
+                    Projection::Field(idx) => match base_ty {
+                        Ty::Struct(sty) => SIR.ty(&sty.fields.tys[usize::try_from(idx).unwrap()]),
+                        Ty::Tuple(tty) => {
+                            let sidx = usize::try_from(idx).unwrap();
+                            let fty = &tty.fields.tys[sidx];
+                            let sty = SIR.ty(fty);
+                            println!("{} {:?} {}", sidx, fty, sty);
```

Comment:
Forgotten print? If we remove it we can also squash this into a one liner as in the `Struct` case.

Reply:
Yes, that's left over debugging.